src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,141 @@
+import {
+  ChangeEvent,
+  useContext,
+  useEffect,
+} from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { DispatchContext, StateContext } from '../../State/State';
+import { deleteTodo, updateTodo } from '../../api/todos';
+
+type Props = {
+  todo: Todo;
+  setIsEditing?: (v: boolean) => void
+  isLoading?: boolean,
+  setIsLoading?: (v: boolean) => void
+  currentTitle?: string,
+  setCurrentTitle?: (v: string) => void
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  setIsEditing,
+  isLoading,
+  setIsLoading,
+  currentTitle,
+  setCurrentTitle,
+}) => {
+  const { title, completed, id } = todo;
+
+  const { clearAll, isEscapeKeyup } = useContext(StateContext);
+  const dispatch = useContext(DispatchContext);
+
+  useEffect(() => {
+    if (clearAll && completed && setIsLoading) {
+      setIsLoading(true);
+    }
+  }, [clearAll, completed, dispatch, setIsLoading]);
+
+  useEffect(() => {
+    if (isEscapeKeyup && setCurrentTitle && setIsEditing) {
```

Comment:
didn't get why you need `isEscapeKeyup` variable.
Keep `isEditing` state here - it is an individual state of each TodoItem

Reply:
Oh, I see, it's part of `TodoField`

Logic in these components is too complicated. You may try to make it simpler and write it's all in one component